DAOS-5507 daos: add daos filesystem copy

dsikich opened the pull request:
A filesystem copy allows copies to and from POSIX for DAOS->POSIX, POSIX->DAOS, and DAOS->DAOS. This version was created as a serial daos utility, which does not require MPI. 

copy to and from a POSIX filesystem:
        --src=daos://<pool/cont> | <path>
        --dst=daos://<pool/cont> | <path>
                    type is daos, only specified if pool/cont used, not with path

A UNS path does not require a daos:// string prefix, and a container subset can be specified by doing --<src | dst>=daos://<pool/cont>/path/in/cont.

daosbuild1 commented on src/utils/daos.c:
```diff
@@ -933,6 +962,30 @@ call_dfuse_ioctl(char *path, struct dfuse_il_reply *reply)
+	switch (op) {
+	case FS_COPY:
+		if (ap->src == NULL || ap->dst == NULL) {
+			fprintf(stderr, "a source and destination path must be provided\n");
```
(style)  line over 80 characters

daosbuild1 commented on src/utils/daos.c:
```diff
@@ -933,6 +962,30 @@ call_dfuse_ioctl(char *path, struct dfuse_il_reply *reply)
+	default:
+		break;
+	}
+	return rc; 
```
(style)  trailing whitespace

daosbuild1 replied:
```suggestion
	return rc;
```

daosbuild1 commented on src/utils/daos.c:
```diff
@@ -1223,13 +1276,34 @@ do { \
+	" filesystem copy options (copy):\n" \
+	"	--src=daos://<pool/cont> | <path>\n" \
+	"	--dst=daos://<pool/cont> | <path>\n" \
+	"	\t type is daos, only specified if pool/cont used, not with path \n"); \
```
(style)  line over 80 characters
(style)  unnecessary whitespace before a quoted newline

daosbuild1 replied:
```suggestion
	"	\t type is daos, only specified if pool/cont used, not with path\n"); \
```

daosbuild1 commented on src/utils/daos.c:
```diff
@@ -1223,13 +1276,34 @@ do { \
+	"container (cont) commands:\n" \
+	"	  copy		copy to/from POSIX filesystem\n"); \
+	fprintf(stream, "\n"); \
+	fprintf(stream, "use 'daos help fs|filesystem COMMAND' for command specific options\n"); \
```
(style)  line over 80 characters

daosbuild1 commented on src/utils/daos.c:
```diff
@@ -1284,6 +1358,10 @@ help_hdlr(int argc, char *argv[], struct cmd_args_s *ap)
 
 	if (argc <= 2) {
 		FIRST_LEVEL_HELP();
+	} else if (strcmp(argv[2], "filesystem") == 0 || strcmp(argv[2], "fs") == 0) {
```
(style)  line over 80 characters

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -48,6 +49,37 @@
 
 #include "daos_hdlr.h"
 
+#define NUM_DIRENTS 	24
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -48,6 +49,37 @@
 #include "daos_hdlr.h"
 
+#define NUM_DIRENTS 	24
+#define MAX_FILENAME 	256
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -48,6 +49,37 @@
+
+struct fs_copy_dirent {
+	dfs_obj_t	*dir;
+	struct 		dirent ents[NUM_DIRENTS];
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -48,6 +49,37 @@
+};
+
+struct file_dfs {
+	enum 		{POSIX, DAOS} type;
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1262 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	char	*cont_name 	= NULL;
+	int	rc		= 0;
+
+	if (path == NULL || _obj_name == NULL || _cont_name == NULL)
```
(style)  suspect code indent for conditional statements (8, 24)

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1262 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			}
+		} else {
+			char *new_dir = calloc(strlen(cwd) + strlen(cont_name) + 1, sizeof(char));
+			if (new_dir == NULL) {
```
(style)  Missing a blank line after declarations

daosbuild1 reviewed (changes requested):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-4193/1/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

Note: Error annotation limited to the first 31 errors.  Remaining unannotated errors:

[src/utils/daos_hdlr.c:1815](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1815):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:1816](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1816):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:1817](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1817):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:1828](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1828):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:1830](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1830):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:1832](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1832):
(style)  unnecessary whitespace before a quoted newline

[src/utils/daos_hdlr.c:1836](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1836):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:1837](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1837):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:1838](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1838):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:1856](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1856):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:1922](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1922):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:1925](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1925):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:1927](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1927):
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:1936](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1936):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:1972](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1972):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:1973](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1973):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:1974](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1974):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:1975](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1975):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:1976](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1976):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:1983](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L1983):
(style)  unnecessary whitespace before a quoted newline

[src/utils/daos_hdlr.c:2018](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2018):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2049](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2049):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2085](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2085):
(style)  "(foo*)" should be "(foo *)"

[src/utils/daos_hdlr.c:2105](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2105):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2114](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2114):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2142](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2142):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2160](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2160):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2161](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2161):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2162](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2162):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2171](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2171):
(style)  unnecessary whitespace before a quoted newline

[src/utils/daos_hdlr.c:2205](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2205):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2213](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2213):
(style)  else is not generally useful after a break or return

[src/utils/daos_hdlr.c:2221](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2221):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2233](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2233):
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2245](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2245):
(style)  space prohibited between function name and open parenthesis '('

[src/utils/daos_hdlr.c:2251](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2251):
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2260](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2260):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2269](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2269):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2278](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2278):
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2284](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2284):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2286](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2286):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2297](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2297):
(style)  line over 80 characters
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2301](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2301):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2306](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2306):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2311](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2311):
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2317](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2317):
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2320](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2320):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2321](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2321):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2323](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2323):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2328](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2328):
(style)  line over 80 characters
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2330](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2330):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2346](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2346):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2351](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2351):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2359](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2359):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2360](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2360):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2363](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2363):
(style)  line over 80 characters
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2366](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2366):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2369](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2369):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2372](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2372):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2377](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2377):
(style)  line over 80 characters
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2378](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2378):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2382](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2382):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2383](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2383):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2385](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2385):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2389](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2389):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2392](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2392):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2396](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2396):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2398](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2398):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2404](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2404):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2405](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2405):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2410](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2410):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2418](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2418):
(style)  space prohibited between function name and open parenthesis '('

[src/utils/daos_hdlr.c:2425](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2425):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2426](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2426):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2428](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2428):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2430](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2430):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2431](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2431):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2441](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2441):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2447](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2447):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2450](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2450):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2462](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2462):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2468](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2468):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2471](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2471):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2473](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2473):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2482](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2482):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2487](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2487):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2488](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2488):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2499](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2499):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2510](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2510):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2511](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2511):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2512](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2512):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2513](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2513):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2518](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2518):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2519](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2519):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2531](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2531):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2536](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2536):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2537](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2537):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2544](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2544):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2549](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2549):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2554](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2554):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2555](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2555):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2575](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2575):
(style)  do not use assignment in if condition

[src/utils/daos_hdlr.c:2582](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2582):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2602](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2602):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2603](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2603):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2606](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2606):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2618](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2618):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2624](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2624):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2626](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2626):
(style)  line over 80 characters
(style)  Avoid unnecessary line continuations
(style)  unnecessary whitespace before a quoted newline

[src/utils/daos_hdlr.c:2638](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2638):
(style)  line over 80 characters
(style)  Avoid unnecessary line continuations
(style)  unnecessary whitespace before a quoted newline

[src/utils/daos_hdlr.c:2646](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2646):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2659](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2659):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2663](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2663):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2668](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2668):
(style)  code indent should use tabs where possible

[src/utils/daos_hdlr.c:2670](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2670):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2671](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2671):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2672](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2672):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2680](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2680):
(style)  Avoid unnecessary line continuations
(style)  unnecessary whitespace before a quoted newline

[src/utils/daos_hdlr.c:2686](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2686):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2688](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2688):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2693](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2693):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2695](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2695):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2699](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2699):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2703](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2703):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2708](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2708):
(style)  code indent should use tabs where possible

[src/utils/daos_hdlr.c:2710](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2710):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2720](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2720):
(style)  Avoid unnecessary line continuations
(style)  unnecessary whitespace before a quoted newline

[src/utils/daos_hdlr.c:2732](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2732):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2733](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2733):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2734](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2734):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2743](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2743):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2744](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2744):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2747](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2747):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2772](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2772):
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2777](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2777):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2779](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2779):
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2796](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2796):
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2808](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2808):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2826](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2826):
(style)  line over 80 characters

[src/utils/daos_hdlr.c:2846](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.c#L2846):
(style)  trailing whitespace

[src/utils/daos_hdlr.h:91](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.h#L91):
(style)  line over 80 characters

[src/utils/daos_hdlr.h:92](https://github.com/daos-stack/daos/blob/5d83e8ff90c5102be1055520e41a5588ce361166/src/utils/daos_hdlr.h#L92):
(style)  line over 80 characters

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4193/1/execution/node/60/log

daosbuild1 commented on src/utils/daos.c:
```diff
@@ -1223,13 +1277,34 @@ do { \
+	"container (cont) commands:\n" \
+	"	  copy		copy to/from POSIX filesystem\n"); \
+	fprintf(stream, "\n"); \
+	fprintf(stream, "use 'daos help fs|filesystem COMMAND' for options\n"); \
```
(style)  line over 80 characters

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	char	*cont_name = NULL;
+	int	rc = 0;
+
+	if (path == NULL ||_obj_name == NULL || _cont_name == NULL)
```
(style)  spaces required around that '||' (ctx:WxV)

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+open_dfs(struct file_dfs *file_dfs, char *file, int flags, mode_t mode)
+{
+
+	int 	  rc = 0;
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+{
+
+	int 	  rc = 0;
+	int 	  tmp_rc = 0; 
```
(style)  trailing whitespace
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	int 	  rc = 0;
+	int 	  tmp_rc = 0; 
+	dfs_obj_t *parent = NULL;
+	char 	  *name	= NULL;
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			D_GOTO(out, rc = EINVAL);
+		}
+		rc = dfs_mkdir(file_dfs->dfs, parent, name, *mode, 0);
+		if (rc != 0) {	
```
(style)  trailing whitespace

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	return rc;
+}
+
+static int 
```
(style)  trailing whitespace

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+}
+
+static struct dirent*
+readdir_dfs(struct file_dfs *file_dfs, DIR *_dirp) 
```
(style)  trailing whitespace

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+				&dirp->anchor, &dirp->num_ents,
+				dirp->ents);
+		if (rc) {
+			fprintf(stderr, "dfs_readdir failed (%d %s)\n", rc, strerror(rc));
```
(style)  line over 80 characters

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+static int
+stat_dfs(struct file_dfs *file_dfs, const char *path, struct stat *buf)
+{
+	int 		rc = 0;
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+stat_dfs(struct file_dfs *file_dfs, const char *path, struct stat *buf)
+{
+	int 		rc = 0;
+	int 		tmp_rc = 0;
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+{
+	int 		rc = 0;
+	int 		tmp_rc = 0;
+	dfs_obj_t 	*parent = NULL;
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	int 		rc = 0;
+	int 		tmp_rc = 0;
+	dfs_obj_t 	*parent = NULL;
+	char 		*name = NULL;
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	int 		tmp_rc = 0;
+	dfs_obj_t 	*parent = NULL;
+	char 		*name = NULL;
+	char 		*dir_name = NULL;
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	/* execute read operation */
+	d_iov_set(&iov, buf, size);
+	int rc = dfs_read(file_dfs->dfs, file_dfs->obj, &sgl,
+			  file_dfs->offset, &got_size, NULL); 
```
(style)  trailing whitespace

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+static int
+closedir_dfs(DIR *_dirp)
+{
+	struct	fs_copy_dirent *dirp	= (struct fs_copy_dirent*)_dirp;
```
(style)  "(foo*)" should be "(foo *)"

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	if (file_dfs->type == POSIX) {
+		rc = closedir(dirp);
+		/* POSIX returns -1 on error and sets errno
+		 * to the error code */
```
(style)  Block comments use a trailing */ on a separate line

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+static int
+chmod_dfs(struct file_dfs *file_dfs, const char *file, mode_t mode)
+{
+	int 		rc = 0;
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+chmod_dfs(struct file_dfs *file_dfs, const char *file, mode_t mode)
+{
+	int 		rc = 0;
+	int 		tmp_rc = 0;
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+{
+	int 		rc = 0;
+	int 		tmp_rc = 0;
+	dfs_obj_t 	*parent	= NULL;
```
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	return rc;
+}
+
+static int 
```
(style)  trailing whitespace

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	/* check it was opened. */
+	if (!src_dir) {
+		fprintf(stderr, "Cannot open directory '%s': %s\n",
+			dir_name, strerror (errno));
```
(style)  space prohibited between function name and open parenthesis '('

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	}
+
+	/* initialize DAOS anchor */
+	struct fs_copy_dirent *dirp = (struct fs_copy_dirent*)src_dir;
```
(style)  "(foo*)" should be "(foo *)"

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+				d_name, strerror(errno));
+			D_GOTO(out, rc);
+		}
+				
```
(style)  trailing whitespace

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	/* After going through all the entries, close the directory. */
+	if (file_closedir(src_file_dfs, src_dir))
+		fprintf(stderr, "Could not close '%s': %s\n",
+			dir_name, strerror (errno));
```
(style)  space prohibited between function name and open parenthesis '('

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+static int
+fs_copy_connect(struct file_dfs *src_file_dfs,
+		struct file_dfs *dst_file_dfs,
+	   	struct fs_copy_args *fa,
```
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		struct file_dfs *dst_file_dfs,
+	   	struct fs_copy_args *fa,
+		char *sysname,
+	   	daos_cont_info_t *src_cont_info,
```
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		char *sysname,
+	   	daos_cont_info_t *src_cont_info,
+		daos_cont_info_t *dst_cont_info,
+	   	struct duns_attr_t *src_dattr,
```
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	   	daos_cont_info_t *src_cont_info,
+		daos_cont_info_t *dst_cont_info,
+	   	struct duns_attr_t *src_dattr,
+	   	struct duns_attr_t *dst_dattr)
```
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+{
+	/* set defaults for file_dfs struct */
+	file_dfs->type = DAOS;
+	file_dfs->fd = -1; 
```
(style)  trailing whitespace

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-4193/2/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

Note: Error annotation limited to the first 31 errors.  Remaining unannotated errors:

[src/utils/daos_hdlr.c:2575](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2575):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2576](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2576):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2577](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2577):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2582](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2582):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2583](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2583):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2644](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2644):
(style)  do not use assignment in if condition

[src/utils/daos_hdlr.c:2651](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2651):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2671](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2671):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2672](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2672):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2695](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2695):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2698](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2698):
(style)  Avoid unnecessary line continuations
(style)  Avoid line continuations in quoted strings

[src/utils/daos_hdlr.c:2711](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2711):
(style)  Avoid unnecessary line continuations
(style)  Avoid line continuations in quoted strings

[src/utils/daos_hdlr.c:2719](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2719):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2743](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2743):
(style)  code indent should use tabs where possible

[src/utils/daos_hdlr.c:2748](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2748):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2749](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2749):
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2757](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2757):
(style)  Avoid unnecessary line continuations
(style)  unnecessary whitespace before a quoted newline

[src/utils/daos_hdlr.c:2764](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2764):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2766](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2766):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2771](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2771):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2789](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2789):
(style)  code indent should use tabs where possible

[src/utils/daos_hdlr.c:2803](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2803):
(style)  Avoid unnecessary line continuations
(style)  unnecessary whitespace before a quoted newline

[src/utils/daos_hdlr.c:2816](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2816):
(style)  trailing whitespace

[src/utils/daos_hdlr.c:2817](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2817):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2818](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2818):
(style)  please, no space before tabs

[src/utils/daos_hdlr.c:2856](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2856):
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2861](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2861):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2863](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2863):
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2880](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2880):
(style)  Missing a blank line after declarations

[src/utils/daos_hdlr.c:2892](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2892):
(style)  Block comments use a trailing */ on a separate line

[src/utils/daos_hdlr.c:2931](https://github.com/daos-stack/daos/blob/ba8dfc7c0322065b9868c85ff24ddd4201fffc6c/src/utils/daos_hdlr.c#L2931):
(style)  trailing whitespace

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4193/2/execution/node/61/log

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4193/2/execution/node/264/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4193/2/execution/node/283/log

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4193/2/execution/node/299/log

daosbuild1 commented:
Test stage Build on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4193/2/execution/node/350/log

daosbuild1 commented:
Test stage Build on CentOS 7 release completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4193/2/execution/node/338/log

daosbuild1 commented:
Test stage Build on CentOS 7 debug completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4193/2/execution/node/352/log

daosbuild1 commented:
Test stage Build on Ubuntu 20.04 with Clang completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4193/2/execution/node/295/log

daosbuild1 commented:
Test stage Build on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4193/2/execution/node/318/log

daosbuild1 commented:
Test stage Build on Leap 15 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4193/2/execution/node/321/log

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		dst += 7;
+		if (strncmp(dst, "/", 1) == 0) {
+			fprintf(stderr, "cannot parse daos dst type format, "
+				"please use: \n"
```
(style)  unnecessary whitespace before a quoted newline

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		}
+		*src_str_len = strlen(*src_str);
+	} else if (strcmp(src_type, "path") == 0) {
+	        rc = duns_resolve_path(src, src_dattr);
```
(style)  code indent should use tabs where possible

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			/* fs_copy will later check this is a valid src path
+			 * with stat
+			 */
+		  	*src_str = strdup(src);
```
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			 * with stat
+			 */
+		  	*src_str = strdup(src);
+		    	src_file_dfs->type = POSIX;
```
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		}
+		*dst_str_len = strlen(*dst_str);
+	} else if (strcmp(dst_type, "path") == 0) {
+	        rc = duns_resolve_path(dst, dst_dattr);
```
(style)  code indent should use tabs where possible

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		if (rc != 0) {
+			/* fs_copy will later check this is a valid dst
+			 * path with stat
+ 			*/
```
(style)  code indent should use tabs where possible
(style)  please, no space before tabs

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		mode_t tmp_mode_dir = S_IRWXU;
+		/* construct destination directory in DAOS, this needs
+		 * to strip the dirname and only use the basename that is
+		 * specified in the dst argument */
```
(style)  Block comments use a trailing */ on a separate line

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	if (src_file_dfs.type == DAOS) {
+		if (src_str != NULL)
+			free(src_str);
+	} 
```
(style)  trailing whitespace

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	sgl.sg_nr_out = 1;
+	d_iov_set(&iov, buf, size);
+	rc = dfs_write(file_dfs->dfs, file_dfs->obj, &sgl,
+			file_dfs->offset, NULL);
```
```suggestion
		       file_dfs->offset, NULL);
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	}
+	file_dfs->offset += (daos_off_t)size;
+out:
+	return (ssize_t) size;
```
```suggestion
	return (ssize_t)size;
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		int path_length = 0;
+
+		path_length = snprintf(filename, MAX_FILENAME, "%s/%s",
+					dir_name, d_name);
```
```suggestion
				       dir_name, d_name);
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+				d_name, strerror(errno));
+			D_GOTO(out, rc);
+		}
+				
+		path_length = snprintf(dst_filename, MAX_FILENAME,
+					"%s/%s", fs_dst_prefix,
```
```suggestion

		path_length = snprintf(dst_filename, MAX_FILENAME,
				       "%s/%s", fs_dst_prefix,
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+				D_GOTO(out, rc);
+			}
+			rc = file_open(dst_file_dfs, dst_filename, dst_flags,
+					tmp_mode_file);
```
```suggestion
				       tmp_mode_file);
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			/* read from source file, then write to dest file */
+			uint64_t file_length = st.st_size;
+			uint64_t total_bytes = 0;
+			uint64_t buf_size = 64*1024*1024;
```
```suggestion
			uint64_t buf_size = 64 * 1024 * 1024;
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+				uint64_t bytes_left = file_length - total_bytes;
+
+				if (bytes_left < buf_size) {
+					left_to_read = (size_t) bytes_left;
```
```suggestion
					left_to_read = (size_t)bytes_left;
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+						filename);
+					D_GOTO(out, rc = EIO);
+				}
+				size_t bytes_to_write = (size_t) bytes_read;
```
```suggestion
				size_t bytes_to_write = (size_t)bytes_read;
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+				ssize_t bytes_written;
+
+				bytes_written = file_write(dst_file_dfs,
+							dst_filename,
```
```suggestion
							   dst_filename,
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			if (buf != NULL)
+				D_FREE(buf);
+
+			/* reset offests if there is another file to copy */
```
```suggestion
			/* reset offsets if there is another file to copy */
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			 * or src_dirs's parent.
+			 */
+			if ((strcmp(d_name, "..") != 0) &&
+				(strcmp(d_name, ".") != 0)) {
```
```suggestion
			    (strcmp(d_name, ".") != 0)) {
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+				 * with the new path.
+				 */
+				rc = fs_copy(src_file_dfs, dst_file_dfs, path,
+						dfs_prefix_len, fs_dst_prefix);
```
```suggestion
					     dfs_prefix_len, fs_dst_prefix);
```

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-4193/3/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	sgl.sg_nr_out = 1;
+	d_iov_set(&iov, buf, size);
+	rc = dfs_write(file_dfs->dfs, file_dfs->obj, &sgl,
+		file_dfs->offset, NULL);
```
```suggestion
		       file_dfs->offset, NULL);
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		D_GOTO(out, rc = EINVAL);
+	}
+	rc = dfs_open(file_dfs->dfs, parent, name, mode | S_IFREG,
+			flags, 0, 0, NULL, &(file_dfs->obj));
```
```suggestion
		      flags, 0, 0, NULL, &file_dfs->obj);
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+
+static ssize_t
+file_read(struct file_dfs *file_dfs, char *file,
+ void *buf, size_t size)
```
```suggestion
	  void *buf, size_t size)
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		int path_length = 0;
+
+		path_length = snprintf(filename, MAX_FILENAME, "%s/%s",
+				dir_name, d_name);
```
```suggestion
				       dir_name, d_name);
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			D_GOTO(err_src_root, rc);
+		}
+		rc = dfs_mount(fa->src_poh, fa->src_coh, O_RDWR,
+				&(src_file_dfs->dfs));
```
```suggestion
			       &src_file_dfs->dfs);
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			D_GOTO(err_dst_root, rc);
+		}
+		rc = dfs_mount(fa->dst_poh, fa->dst_coh, O_RDWR,
+				&(dst_file_dfs->dfs));
```
```suggestion
			       &dst_file_dfs->dfs);
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+
+static int
+fs_copy_disconnect(struct fs_copy_args *fa,
+		struct file_dfs *src_file_dfs,
```
```suggestion
		   struct file_dfs *src_file_dfs,
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		/* check if this copy is from root of DFS cont */
+		*src_str = strdup("/");
+		*src_dfs_path = strdup(src);
+		rc = fs_copy_parse_uuids(src, &src_saveptr, &(fa->src_p_uuid),
+					 &(fa->src_c_uuid), &src_pool,
```
```suggestion
		rc = fs_copy_parse_uuids(src, &src_saveptr, &fa->src_p_uuid,
					 &fa->src_c_uuid, &src_pool,
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		*dst_str = strdup("/");
+		*dst_dfs_path = strdup(dst);
+		rc = fs_copy_parse_uuids(dst, &dst_saveptr,
+					 &(fa->dst_p_uuid), &(fa->dst_c_uuid),
```
```suggestion
					 &fa->dst_p_uuid, &fa->dst_c_uuid,
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		D_GOTO(out, rc);
+	}
+	rc = fs_copy_connect(&src_file_dfs, &dst_file_dfs, &fa,
+			   ap->sysname, &src_cont_info, &dst_cont_info,
```
```suggestion
			     ap->sysname, &src_cont_info, &dst_cont_info,
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		if (rc != EEXIST && rc != 0)
+			D_GOTO(out_disconnect, rc);
+		rc = fs_copy(&src_file_dfs, &dst_file_dfs,
+				src_str, src_str_len, dst_str);
```
```suggestion
			     src_str, src_str_len, dst_str);
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		if (rc != EEXIST && rc != 0)
+			D_GOTO(out_disconnect, rc);
+		rc = fs_copy(&src_file_dfs, &dst_file_dfs,
+				src_str, src_str_len, dst_str);
```
```suggestion
			     src_str, src_str_len, dst_str);
```

daosbuild1 commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		if (rc != EEXIST && rc != 0)
+			D_GOTO(out_disconnect, rc);
+		rc = fs_copy(&src_file_dfs, &dst_file_dfs,
+				src_str, src_str_len, dst_str);
```
```suggestion
			     src_str, src_str_len, dst_str);
```

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-4193/16/execution/node/61/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

dsikich commented:
@mchaarawi @daltonbohning took a few tries for me to get this through check patch, but it is now. I restarted the CI since it looks like it got stopped for some reason.

daltonbohning commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+{
+	int rc = 0;
+
+	if (src_file_dfs->type == DAOS) {
+		rc = dfs_umount(src_file_dfs->dfs);
+		if (rc != 0) {
+			fprintf(stderr, "failed to unmount source (%d)\n", rc);
+			D_GOTO(out, rc);
+		}
+		rc = daos_cont_close(fa->src_coh, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to close source "
+				"container (%d)\n", rc);
+			D_GOTO(out, rc);
+		}
+		rc = daos_pool_disconnect(fa->src_poh, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to disconnect from source "
+				"pool "DF_UUIDF ": %s (%d)\n",
+				DP_UUID(fa->src_p_uuid), d_errdesc(rc), rc);
+			D_GOTO(out, rc);
+		}
```
@dsikich When _**mounting**_ and _**opening**_ some handles, I think it makes sense to `D_GOTO(out, rc);`.
But when _**unmounting**_ and _**closing**_ some handles, I think that if we fail to unmount/close the source, then we should probably still try to unmount/close the destination, so we don't leave those handles open.
@mchaarawi What do you think?

mchaarawi replied:
sure but im ok to handle this in a different PR

daltonbohning commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	}
+
+	/* check for src DAOS pool/cont or UNS path */
+	if (strcmp(src_type, "daos") == 0) {
```
@dsikich I wonder if we should abstract this parsing to a function, to avoid essentially duplicating the logic for parsing the src and dst?

mchaarawi replied:
we should be using the gurt macros (D_*) for str functions; for correct memory tracing, but im ok that being done in a different PR. please keep track of this in a ticket or whatever.

dsikich replied:
@mchaarawi I do not see a macro for strcmp in gurt/common.h? Only for strndup (D_STRNDUP)

daltonbohning reviewed (approved):
@dsikich Functionally, this is passing the basic test cases - I just had some comments about error checking, etc.
I think this is probably a good first pass.
(Update: I'm okay with these things being addressed in a separate PR)

dsikich commented:
@daltonbohning This is getting to be a little too much back and forth over this fs copy PR. If there is still more to implement for filesystem copy in this PR I'd like to close this completely and we can work out on the side explicitly what this ticket will cover, and I can open a new one. Anything else, I think it should go into another ticket. It is getting hard for me to keep track of honestly.

daltonbohning commented:
> @daltonbohning This is getting to be a little too much back and forth over this fs copy PR. If there is still more to implement for filesystem copy in this PR I'd like to close this completely and we can work out on the side explicitly what this ticket will cover, and I can open a new one. Anything else, I think it should go into another ticket. It is getting hard for me to keep track of honestly.

@dsikich I'm okay with these things being done in a separate PR. I mostly just wanted to make sure we had note of them so we don't forget.

dsikich commented:
> > @daltonbohning This is getting to be a little too much back and forth over this fs copy PR. If there is still more to implement for filesystem copy in this PR I'd like to close this completely and we can work out on the side explicitly what this ticket will cover, and I can open a new one. Anything else, I think it should go into another ticket. It is getting hard for me to keep track of honestly.
> 
> @dsikich I'm okay with these things being done in a separate PR. I mostly just wanted to make sure we had note of them so we don't forget.

@daltonbohning yeah its no problem. I'm just wanting to get better at specifying requirements up front so that I don't have to implement new stuff during the middle of the PR, which makes it long and harder for me to keep track of things. If its not ready I just want to avoid submitting anything, that was my bad :-)

mchaarawi commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			   &dst_file_dfs, &src_dattr,
+			   &dst_dattr, &src_str,
+			   &src_dfs_path, &src_str_len,
+			   &dst_str, &dst_dfs_path,
```
i see src/dst_str  freed, but where is src/dst_dfs_path freed?

mchaarawi commented:
> > @daltonbohning This is getting to be a little too much back and forth over this fs copy PR. If there is still more to implement for filesystem copy in this PR I'd like to close this completely and we can work out on the side explicitly what this ticket will cover, and I can open a new one. Anything else, I think it should go into another ticket. It is getting hard for me to keep track of honestly.
> 
> @dsikich I'm okay with these things being done in a separate PR. I mostly just wanted to make sure we had note of them so we don't forget.

to offer my perspective here, i think it is essential to be critical in reviews and go back and forth multiple times if the PR is not ready to land as this was at first. TBH there was a lot of issues and catching them all in one go is not feasible. im pretty sure there are still issues that we haven't caught, as there are in many other places in the code.
the idea of reviews is to be critical and find all possible bugs and fix them. deferring to follow on PRs should be mostly for features and not bugs / leaks. it's better to iterate multiple times rather than having users report bugs especially since we are pushing this to master and not to a private branch.

dsikich commented:
> > > @daltonbohning This is getting to be a little too much back and forth over this fs copy PR. If there is still more to implement for filesystem copy in this PR I'd like to close this completely and we can work out on the side explicitly what this ticket will cover, and I can open a new one. Anything else, I think it should go into another ticket. It is getting hard for me to keep track of honestly.
> > 
> > 
> > @dsikich I'm okay with these things being done in a separate PR. I mostly just wanted to make sure we had note of them so we don't forget.
> 
> to offer my perspective here, i think it is essential to be critical in reviews and go back and forth multiple times if the PR is not ready to land as this was at first. TBH there was a lot of issues and catching them all in one go is not feasible. im pretty sure there are still issues that we haven't caught, as there are in many other places in the code.
> the idea of reviews is to be critical and try to find all possible bugs and fix them. deferring to follow on PRs should be mostly for features and not bugs / leaks. it's better to iterate multiple times rather than having users report bugs especially since we are pushing this to master and not to a private branch.

@mchaarawi @daltonbohning I agree, I do think I submitted this PR too soon looking back. I agree fixing bugs/leaks here makes sense. I was mostly referring to some of the newer implementations I added in the PR (the interface, etc), which also just goes back to submitting this PR too soon for this much code.

daltonbohning commented:
> to offer my perspective here, i think it is essential to be critical in reviews and go back and forth multiple times if the PR is not ready to land as this was at first. TBH there was a lot of issues and catching them all in one go is not feasible. im pretty sure there are still issues that we haven't caught, as there are in many other places in the code.
> the idea of reviews is to be critical and try to find all possible bugs and fix them. deferring to follow on PRs should be mostly for features and not bugs / leaks. it's better to iterate multiple times rather than having users report bugs especially since we are pushing this to master and not to a private branch.

@dsikich I will defer to @mchaarawi's judgement here, since he's generally more experienced - specifically in the DAOS repo.

dsikich commented:
> > to offer my perspective here, i think it is essential to be critical in reviews and go back and forth multiple times if the PR is not ready to land as this was at first. TBH there was a lot of issues and catching them all in one go is not feasible. im pretty sure there are still issues that we haven't caught, as there are in many other places in the code.
> > the idea of reviews is to be critical and try to find all possible bugs and fix them. deferring to follow on PRs should be mostly for features and not bugs / leaks. it's better to iterate multiple times rather than having users report bugs especially since we are pushing this to master and not to a private branch.
> 
> @dsikich I will defer to @mchaarawi's judgement here, since he's generally more experienced - specifically in the DAOS repo.

@daltonbohning @mchaarawi makes sense to me. I learned a lot from this PR, so in the future I want to have it further along before requesting reviews.

johannlombardi commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	}
+out:
+		if (f1)
+			free(f1);
```
alignment?

johannlombardi commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		if (f1)
+			free(f1);
+		if (f2)
+			free(f2);
```
alignment?

johannlombardi commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	}
+	if (num_bytes_written < 0) {
+		fprintf(stderr, "write error on %s type=%d\n",
+			file, file_dfs->type);
```
Should maybe print errno or return error?

johannlombardi commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+{
+	int rc;
+	d_iov_t	iov;
+	d_sg_list_t sgl;
```
please align var names.

johannlombardi commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		}
+		int src_len = strlen(*src_dfs_path);
+		int src_pool_cont_len = (strlen(src_pool) +
+					strlen(src_cont) + 1);
```
please don't define var in the middle of code.

johannlombardi commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		}
+		int dst_len = strlen(*dst_dfs_path);
+		int dst_pool_cont_len = (strlen(dst_pool) +
+					strlen(dst_cont) + 1);
```
ditto.

johannlombardi commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	}
+
+	/* set paths based on file type for source and destination */
+	char *name = NULL, *dname = NULL;
```
ditto.

johannlombardi commented on src/utils/daos_hdlr.c:
```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			free(dst_str);
+	}
+	return rc;
+}
```
Quite a lot of code here, should probably be moved to a separate file eventually.

dsikich replied:
@johannlombardi I think after the dfs wrapper functions are moved that should reduce the amount of code added to daos_hdlr.c significantly. Link to that is here: https://jira.hpdd.intel.com/browse/DAOS-6098

johannlombardi commented on src/utils/daos.c:
```diff
@@ -667,6 +687,16 @@ common_op_parse_hdlr(int argc, char *argv[], struct cmd_args_s *ap)
+			D_STRNDUP(ap->dst, optarg, strlen(optarg));
+			if (ap->dst == NULL)
+				D_GOTO(out_free, rc = RC_NO_HELP);
+			break;
```
We need to update the man page too under doc/man/man8/daos.8

johannlombardi reviewed (commented):
Some cleanup is required and can be done in a follow-on patch.

dsikich commented:
@mchaarawi @daltonbohning ticket for addressing comments is here: https://jira.hpdd.intel.com/browse/DAOS-6428

I can start/address these today and tomorrow.

johannlombardi commented:
@dsikich please address my comment above too.

dsikich commented:
@johannlombardi Working on them now. Thanks!
